www/lobby/id.ts

```diff
@@ -1,16 +1,42 @@
-import { Router, Request, Response } from 'express';
+import { Router, Request, Response, NextFunction } from 'express';
+// import { nextTick } from 'process';
 import { User } from '../../lib';
 import { Lobby } from '../../lib/lobby';
+import { validateJwt, validateLobbyJwt } from '../../utils/jwt';
 
 export const lobby_id_router = Router();
 
+const joinLobby = async (req: Request, res: Response) => {
+  // check if token is valid
+  const lobbyId: string = req.body.lobbyId;
+  const lobby = await Lobby.fromId(lobbyId);
+
+  if (!lobby || lobbyId != req.params.id) return res.status(406).end('lobby token is invalid');
+
+  //add user to lobby
+  lobby?.addUser(req.body.userId);
+  res.status(200).send({...lobby.getClientResponse()});
+};
+
 // Join a lobby
 /**
  * Body Params: id, lobbyToken, refreshToken
  */
-lobby_id_router.post('/:id', async (req: Request, res: Response) => {
-  res.status(200).json('placeholder post');
-});
+lobby_id_router.post('/:id', validateJwt, async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
in my opinion, you have to many functions going on here.

If you were following this implementation, I would recommend putting the **finishing** function as the one that actually shows here. Because it is the last function that runs! It doesn't really make sense to move it outside the endpoint declaration.

I think a better design for this endpoint is the following:

```suggestion
// for validateLobbyJwt, add an undefined check and then run next() and return

lobby_id_router.post('/:id', validateJwt, validateLobbyJwt, async (req: Request, res: Response) => {
  const userId: string = req.body.id;
  const lobbyId: string = req.param.id;
  const decodedLobby: string | undefined = req.body.lobbyId;
  
  const verified = await verifyIds(userId, lobbyId);
  if (!verified) return res.status(404).json('User or Lobby not found in database').end();
  const [user, lobby] = verified;
  
  // Let's make a helper function in Lobby to do this for us (param should accept both string and User)
  const isParticipant = lobby.participants.include(user);
  if(!isParticpant && lobbyId !== decodedLobbyId) {
     // return 406
  }
  if (!isParticipant) {
    lobby.add(user);
  }
  // return 200 + data
});
```


Use trevor's helper function here:

https://github.com/UCLA-Creative-Labs/aurgy-backend/blob/404dfa748331313f4dbd38894be558aa003eea1a/www/lobby/:id.ts#L78-L84

Reply:
Ok I implemented this, let me know if I did anything wrong